### PR TITLE
fix new Sample(SparseTensor, DenseTensor) will throw exception.

### LINKED ISSUE #2258: new Sample(SparseTensor, DenseTensor) will throw exception.
@michellemay raised this issue in #979

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

fix new Sample(SparseTensor, DenseTensor) will throw exception.

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed #2258

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala [hunk 1]
```diff
@@ -224,11 +224,13 @@ private[tensor] class SparseTensor[@specialized(Float, Double) T: ClassTag](
   }
 
   override def isContiguous(): Boolean = {
-    throw new UnsupportedOperationException(s"SparseTensor: Unimplemented method")
+    // Sparse Tensor's storage is always contiguous
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala]: why?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala [hunk 2]
```diff
@@ -224,11 +224,13 @@ private[tensor] class SparseTensor[@specialized(Float, Double) T: ClassTag](
...
+    true
   }
 
   override def contiguous(): Tensor[T] = {
-    throw new UnsupportedOperationException(s"SparseTensor: Unimplemented method")
+    // Sparse Tensor's storage is always contiguous
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/SparseTensor.scala]: why?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/dataset/SampleSpec.scala [hunk 3]
```diff
@@ -61,6 +62,32 @@ class SampleSpec extends FlatSpec with Matchers {
...
 
+  "create Sample" should "work fine" in {
+    val st1 = Tensor.sparse(Tensor.range(1, 10, 1))
+    val st2 = Tensor.sparse(Tensor.range(1, 10, 1))
+    val dt1 = Tensor.range(1, 10, 1)
+    val dt2 = Tensor.range(1, 10, 1)
+    val label1 = Tensor(1).fill(1)
+    val label2 = Tensor(1).fill(2)
+
+    Sample(st1)
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/dataset/SampleSpec.scala]: does this cover all sample factory methods?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala [hunk 4]
```diff
@@ -338,8 +338,6 @@ object Sample {
   def apply[T: ClassTag](
         featureTensor: Tensor[T],
         labelTensor: Tensor[T])(implicit ev: TensorNumeric[T]) : Sample[T] = {
-    require(featureTensor.isContiguous(), "featureTensor is not contiguous")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala]: if sparse tensor is contiguous, why remove this?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala [hunk 5]
```diff
@@ -350,7 +348,6 @@ object Sample {
   def apply[T: ClassTag](
         featureTensor: Tensor[T],
         label: T)(implicit ev: TensorNumeric[T]) : Sample[T] = {
-    require(featureTensor.isContiguous(), "featureTensor is not contiguous")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala]: same with above

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala [hunk 6]
```diff
@@ -381,7 +378,6 @@ object Sample {
 
   def apply[T: ClassTag](
         featureTensor: Tensor[T])(implicit ev: TensorNumeric[T]) : Sample[T] = {
-    require(featureTensor.isContiguous(), "featureTensor is not contiguous")
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala]: same with above

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala (hunk 4 above)
Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala]: it's duplicated checking, will check contiguous in ArraySample.apply

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/dataset/SampleSpec.scala (hunk 3 above)
Comment (qiuxin2012) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/dataset/SampleSpec.scala]: One missing

Comment (qiuxin2012) [reply spark/dl/src/test/scala/com/intel/analytics/bigdl/dataset/SampleSpec.scala]: Added

**Comment by yiheng** [comment]
where do you bypass the sparse tensor?

**Comment by qiuxin2012** [comment]
jenkins passed.

**Pull request merged**